packages/opentelemetry-core/src/common/logging-error-handler.ts

```diff
@@ -14,26 +14,16 @@
  * limitations under the License.
  */
 
-import {
-  Logger,
-  Exception,
-  DiagLogLevel,
-  DiagConsoleLogger,
-  createLogLevelDiagLogger,
-} from '@opentelemetry/api';
+import { diag, Exception } from '@opentelemetry/api';
 import { ErrorHandler } from './types';
 
 /**
  * Returns a function that logs an error using the provided logger, or a
  * console logger if one was not provided.
- * @param {Logger} logger
  */
-export function loggingErrorHandler(logger?: Logger): ErrorHandler {
-  logger =
-    logger ??
-    createLogLevelDiagLogger(DiagLogLevel.ERROR, new DiagConsoleLogger());
+export function loggingErrorHandler(): ErrorHandler {
   return (ex: Exception) => {
-    logger!.error(stringifyException(ex));
```

Comment:
i think we could remove the `loggingErrorHandler` so we only the use the diag logger in a future PR

Reply:
No I think we should keep it. The global error handler serves a different (though related) function and having an option to easily push those errors to logs is probably what most people will want most of the time.

Reply:
Well its purpose was for internal components (exporters etc) so i'm not sure why we could not use the diag logger instead ?

Reply:
This gives users the option to trigger some behavior on an unexpected instrumentation failure like restarting the service or preventing startup or something.